### DISCUSSION TITLE
Troubleshoot logflare analytics 401 Unauthorized response

### DISCUSSION THREAD

**Discussion by jeffmccune** [supabase/supabase#20065] [Questions] [answered]
## Questions

 1. How should / could I troubleshoot 401 Unauthorized response bodies from the Next.js studio service backend, received by the browser?
 2. How can I enable request logging / debug logging on the studio service?
 3. How can I find the request id returned in the 401 response body?
 4. Are there better techniques to troubleshoot the studio service?

## Context

I have a self-hosted setup in both kubernetes and docker-compose.  The kubernetes deployment is a faithful translation of the current official [docker-compose](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/docker-compose.yml#L296-L336) file in the main branch.  Nearly everything is working with full parity between the two, except logs, which I'm working on presently.  I have both running simultaneously to compare behavior.

## Expected behavior

In the docker compose, fetching `http://localhost:8000/project/default/logs/edge-logs?ite=&its=` correctly returns the Kong API gateway logs.

## Actual behavior

In my kubernetes cluster, fetching the same path, `https://api.example.com/project/default/logs/edge-logs?ite=&its=` returns:

```json
{
  "code": 401,
  "message": "Unauthorized",
  "requestId": "4561f640-f54a-4464-9ad6-47a434c596ec"
}
```

I can see the fetch request in the browser, and I can see this request makes it through the Kong API gateway and is destined for the studio service:

```
k logs -l app.kubernetes.io/name=kong -f | grep edge-logs
```

```
127.0.0.6 - - [30/Dec/2023:16:28:27 +0000] "GET /api/projects/default/analytics/endpoints/logs.all?project=default&sql=%0ASELECT... HTTP/1.1" 200 98 "https://api.example.som/project/default/logs/edge-logs?ite=&its=" "Mozilla/5.0"
```

I assume this request is sent to the studio given [this route](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/volumes/api/kong.yml#L210-L223):

```yaml
services:
  ## Protected Dashboard - catch all remaining routes
  - name: dashboard
    _comment: 'Studio: /* -> http://studio:3000/*'
    url: http://studio:3000/
    routes:
      - name: dashboard-all
        strip_path: true
        paths:
          - /
    plugins:
      - name: cors
      - name: basic-auth
        config:
          hide_credentials: true
```

However, I'm struggling to figure out why the request is unauthorized.  The studio logs are empty in both docker-compose and kubernetes:

```
❯ docker logs supabase-studio
  ▲ Next.js 13.5.3
  - Local:        http://8bee876c6fa4:3000
  - Network:      http://172.31.0.12:3000

 ✓ Ready in 408ms
```

```
❯ k logs -l app.kubernetes.io/name=studio
  ▲ Next.js 13.5.3
  - Local:        http://studio-74f55c4b8-92tbk:3000
  - Network:      http://10.65.246.88:3000

 ✓ Ready in 104ms
```

Lastly, I've searched through all of the services container log outputs and I can't find that request id (`4561f640-f54a-4464-9ad6-47a434c596ec`) anywhere.  Is this expected?

Please advise, thank you!

**Comment by jeffmccune** [comment]
I verified the request is going from kong to studio by incorporating the studio pod into the service mesh.  This allows me to observe the request id from kong to studio to analytics, but it's still unclear why the request is unauthorized.

```
k logs -l app.kubernetes.io/name=studio -c istio-proxy | grep -i cac80416-b6f0-4b3f-a889-e079244d175d | jq .
```

```json
{
  "protocol": "HTTP/1.1",
  "authority": "analytics:4000",
  "route_name": "default",
  "upstream_local_address": "10.65.200.220:53792",
  "bytes_sent": 24,
  "requested_server_name": null,
  "upstream_host": "10.65.246.126:4000",
  "start_time": "2023-12-30T20:49:23.262Z",
  "user_agent": "node",
  "x_forwarded_for": null,
  "client_ip": null,
  "path": "/api/endpoints/query/logs.all?project=default&sql=%0ASELECT%0A--+log-event-chart%0A++timestamp_trunc%28t.timestamp%2C+minute%29+as+timestamp%2C%0A++count%28t.timestamp%29+as+count%0AFROM%0A++edge_logs+t%0A++cross+join+unnest%28metadata%29+as+m%0A++cross+join+unnest%28m.request%29+as+request%0A++cross+join+unnest%28m.response%29+as+response%0A++where+t.timestamp+%3E+%272023-12-30T13%3A00%3A00.000Z%27%0AGROUP+BY%0Atimestamp%0AORDER+BY%0A++timestamp+ASC%0A++&ref=default&project_tier=ENTERPRISE",
  "method": "GET",
  "upstream_cluster": "outbound|4000||analytics.dev-jeff.svc.cluster.local",
  "response_flags": "-",
  "upstream_transport_failure_reason": null,
  "bytes_received": 0,
  "downstream_remote_address": "10.65.200.220:45902",
  "downstream_local_address": "10.64.27.78:4000",
  "response_code_details": "via_upstream",
  "status_code": 401,
  "request_id": "cac80416-b6f0-4b3f-a889-e079244d175d",
  "duration": 6,
  "upstream_service_time": null
}
```

**Comment by jeffmccune** [comment]
I've verified it's definitely logflare by enabling debug logs with `LOGFLARE_LOG_LEVEL=debug` but I'm still not sure why logflare responds with a 401.

```
22:18:53.330 [debug] GET /api/endpoints/query/logs.all

22:18:53.331 [debug] Processing with LogflareWeb.EndpointsController.query/2
  Parameters: %{"iso_timestamp_start" => "2023-12-30T21:00:00.000Z", "project" => "default", "project_tier" => "ENTERPRISE", "ref" => "default", "sql" => "select id, timestamp, event_message, request.method, request.path, response.status_code\n  from edge_logs\n  cross join unnest(metadata) as m\n  cross join unnest(m.request) as request\n  cross join unnest(m.response) as response\n  \n  order by timestamp desc\n  limit 100\n  ", "token_or_name" => "logs.all"}
  Pipelines: [:api, :require_endpoint_auth]

22:18:53.334 [debug] QUERY OK source="oauth_access_tokens" db=1.6ms queue=1.6ms idle=840.4ms
SELECT o0."id", o0."token", o0."refresh_token", o0."expires_in", o0."revoked_at", o0."scopes", o0."previous_refresh_token", o0."resource_owner_id", o0."application_id", o0."description", o0."inserted_at", o0."updated_at" FROM "oauth_access_tokens" AS o0 WHERE (o0."token" = $1) ["REDACTED"]

22:18:53.338 [debug] QUERY OK source="users" db=1.4ms queue=1.2ms idle=844.3ms
SELECT u0."id", u0."email", u0."provider", u0."token", u0."api_key", u0."old_api_key", u0."email_preferred", u0."name", u0."image", u0."email_me_product", u0."admin", u0."phone", u0."bigquery_project_id", u0."bigquery_dataset_location", u0."bigquery_dataset_id", u0."bigquery_udfs_hash", u0."bigquery_processed_bytes_limit", u0."api_quota", u0."valid_google_account", u0."provider_uid", u0."company", u0."billing_enabled", u0."endpoints_beta", u0."preferences", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."api_key" = $1) ["REDACTED"]

22:18:53.341 [debug] QUERY OK source="endpoint_queries" db=1.2ms decode=0.1ms queue=1.4ms idle=847.6ms
SELECT e0."id", e0."token", e0."name", e0."query", e0."language", e0."source_mapping", e0."sandboxable", e0."cache_duration_seconds", e0."proactive_requerying_seconds", e0."max_limit", e0."enable_auth", e0."user_id", e0."sandbox_query_id", e0."inserted_at", e0."updated_at" FROM "endpoint_queries" AS e0 WHERE ((e0."name" = $1) AND (e0."user_id" = $2)) ["logs.all", 1]

22:18:53.342 [debug] Sent 401 in 11ms
```
```

**Comment by jeffmccune** [comment]
I notice the `endpoint_queries` table is empty, which might explain the 401 Unauthorized response.  I also see this error when the analytics pod starts up, which explains why the endpoint queries table is empty.  I'm curious what's causing this error, I don't see it in the docker-compose setup.  Could it be related to changing the LOGFLARE_API_KEY?

```
23:04:11.994 [error] Task #PID<0.8065.0> started from Logflare.Supervisor terminating
** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{name: "logs.all", query: "with retention as (\n  select (\n    CASE\n      WHEN @project_tier = 'FREE' THEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 day)\n      WHEN @project_tier = 'PRO' THEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 7 day)\n      WHEN (@project_tier = 'PAYG' OR @project_tier = 'ENTERPRISE') THEN TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 90 day) \n      ELSE  TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 day) \n    END\n  ) as date\n),\n\nedge_logs as (\nselect \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata \nfrom retention, `cloudflare.logs.prod` as t\n  cross join unnest(metadata) as m\nwhere\n  -- order of the where clauses matters\n  -- project then timestamp then everything else\n  t.project = @project\n  AND CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN COALESCE(@iso_timestamp_end, '') = '' THEN TRUE ELSE cast(t.timestamp as timestamp) <= cast(@iso_timestamp_end as timestamp) END\n  AND cast(t.timestamp as timestamp) > retention.date\norder by\n  cast(t.timestamp as timestamp) desc\n),\n\npostgres_logs as (\n  select \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata\nfrom retention, `postgres.logs` as t\nwhere\n  -- order of the where clauses matters\n  -- project then timestamp then everything else\n  t.project = @project\n  AND CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN COALESCE(@iso_timestamp_end, '') = '' THEN TRUE ELSE cast(t.timestamp as timestamp) <= cast(@iso_timestamp_end as timestamp) END\n  AND cast(t.timestamp as timestamp) > retention.date\n  order by cast(t.timestamp as timestamp) desc\n),\n\nfunction_edge_logs as (\nselect \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata \nfrom retention, `deno-relay-logs` as t\n  cross join unnest(t.metadata) as m\nwhere\n  CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN COALESCE(@iso_timestamp_end, '') = '' THEN TRUE ELSE cast(t.timestamp as timestamp) <= cast(@iso_timestamp_end as timestamp) END\n  and m.project_ref = @project\n  AND cast(t.timestamp as timestamp) > retention.date\norder by cast(t.timestamp as timestamp) desc\n),\n\nfunction_logs as (\nselect \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata \nfrom retention, `deno-subhosting-events` as t\n  cross join unnest(t.metadata) as m\nwhere\n  -- order of the where clauses matters\n  -- project then timestamp then everything else\n  m.project_ref = @project\n  AND CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN COALESCE(@iso_timestamp_end, '') = '' THEN TRUE ELSE cast(t.timestamp as timestamp) <= cast(@iso_timestamp_end as timestamp) END\n  AND cast(t.timestamp as timestamp) > retention.date\norder by cast(t.timestamp as timestamp) desc\n),\n\nauth_logs as (\nselect \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata \nfrom retention, `gotrue.logs.prod` as t\n  cross join unnest(t.metadata) as m\nwhere\n  -- order of the where clauses matters\n  -- project then timestamp then everything else\n  -- m.project = @project\n  t.project = @project\n  AND CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN COALESCE(@iso_timestamp_end, '') = '' THEN TRUE ELSE cast(t.timestamp as timestamp) <= cast(@iso_timestamp_end as timestamp) END\n  AND cast(t.timestamp as timestamp) > retention.date\norder by cast(t.timestamp as timestamp) desc\n),\n\nrealtime_logs as (\nselect \n  t.timestamp,\n  t.id, \n  t.event_message, \n  t.metadata \nfrom retention, `realtime.logs.prod` as t\n  cross join unnest(t.metadata) as m\nwhere\n  m.project = @project \n  AND CASE WHEN COALESCE(@iso_timestamp_start, '') = '' THEN  TRUE ELSE  cast(t.timestamp as timestamp) > cast(@iso_timestamp_start as timestamp) END\n  AND CASE WHEN" <> ..., sandboxable: true, cache_duration_seconds: 0}, errors: [query: {"can't find source postgres.logs", []}], data: #Logflare.Endpoints.Query<>, valid?: false>}
    (logflare 1.5.13) lib/logflare/single_tenant.ex:198: anonymous fn/2 in Logflare.SingleTenant.create_supabase_endpoints/0
    (elixir 1.15.7) lib/enum.ex:1693: Enum."-map/2-lists^map/1-1-"/2
    (logflare 1.5.13) lib/logflare/single_tenant.ex:197: Logflare.SingleTenant.create_supabase_endpoints/0
    (logflare 1.5.13) lib/logflare/application.ex:273: Logflare.Application.startup_tasks/0
    (elixir 1.15.7) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
Function: #Function<3.93554717/0 in Logflare.Application.get_children/1>
    Args: []

23:04:12.154 [debug] starting init task

```

**Reply by jeffmccune** [reply]
The key piece of information seems to be `errors: [query: {"can't find source postgres.logs", []}]`. 

What is this source and how does it get created in the official docker-compose setup?

Here's [the code](https://github.com/Logflare/logflare/blob/9b90f186e8ad4209eb41f2bee24ae34bc5bef19a/lib/logflare/single_tenant.ex#L198C38-L198C38)

I think this has something to do with [this comment](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/volumes/logs/vector.yml#L198-L201)

    # We must route the sink through kong because ingesting logs before logflare is fully initialised will
    # lead to broken queries from studio. This works by the assumption that containers are started in the
    # following order: vector > db > logflare > kong
    uri: 'http://kong:8000/analytics/v1/api/logs?source_name=postgres.logs&api_key=your-super-secret-and-long-logflare-key'

**Comment by williamoverton** [comment]
Did you ever figure this out?

**Comment by bayerlse** [comment]
Also having this error :/

**Comment by etherealmuse** [comment]
@jeffmccune  getting same issue but with Database, did you have trouble with Database part too or just logs ?

below is from kong pod 

```logs
"POST /pg/query HTTP/1.1" 401 52 "-" "node"
"GET /pg/schemas HTTP/1.1" 401 52 "-" "node"
"GET /pg/tables?include_columns=true&included_schemas=public&ref=default HTTP/1.1" 401 52 "-" "node"
"GET /pg/publications HTTP/1.1" 401 52 "-" "node"
"GET /pg/types HTTP/1.1" 401 52 "-" "node"
"GET /pg/extensions HTTP/1.1" 401 52 "-" "node"
```

**Reply by Ziinc** [reply]
This seems to be unrelated.

**Comment by jeffmccune** [comment]
Never figured this out unfortunately, mainly because I hit my time box limit working on this issue.

**Comment by Ziinc** [comment, marked as answer]
Hi, dev of logflare here. The issue as to why the queries are failing is due to the seeding sources not completing before queries are getting made. Somehow or the other, the necessary supabase log sources are not getting created. 

A few things that we can determine from your analysis:
1. role used definitely has enough perms to query the db, so it isn't a perms issue
2. if the k8s config is translated correctly, there should be 2 boolean flags [set here](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/docker-compose.yml#L324) that will enable the startup seeding.

Some things that I can advise to help with debugging this:
1. ensure that there are [health checks](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/docker-compose.yml#L299) set up for the service, as it will only return healthy if there is data seeded.
2. ensure that the [correct stable version](https://github.com/supabase/supabase/blob/8e4a094f2639202c725d27409d6585e7026b1d6e/docker/docker-compose.yml#L298) of logflare is in use which will correspond with what is used on CLI
3. if you navigate to `localhost:4000/dashboard`, you should see the necessary seeded sources (the ui won't work beyond that), but it would help with verifying that the seeding is done correctly.


Edit 10 sept: Please try to use v1.8.9, as it contains a patch relating to stale PG config.

**Reply by AntonOfTheWoods** [reply]
@Ziinc , is there any plan to make the logs/errors any more meaningful and explicit? To give any usable documentation on how to get things set up outside of just reusing the provided `compose` file?

It seems lots and lots of us are trying to get things working properly in `kubernetes` and this is the one that many of us seem to stumble on. The `supabase-community` helm chart actually does have *some* of the logs appearing in the `studio` interface but that chart points to a single db `deployment` without persistence for both the main db and the logs... So it's really utterly unusable, and I have been completely incapable of getting logs in the `studio` in `kube` using anything else. By trying various things I can get log data going into tables but there doesn't seem to be any way to query either from the `studio` or from the `:4000/dashboard`.

At the very least, there seems to be a certain number of extensions that need to be enabled but I couldn't find anything more than "you are invited to use the docker compose". It would be great to know what we need to put, and whether that has changed in later versions (I can also get values in the db with 1.6.4, but even the community chart won't show logs in the `studio` with that...).

I personally have what looks to be an identical `dashboard` (with endpoints also) for the working compose/community chart and my own, and get the same back from the health check.

Thanks for the great work!

**Reply by Ziinc** [reply]
Can you try with v1.8.9? We have recently fixed a bug relating to stale PG backend configuration where if there is a password change it now gets correctly adjusted at runtime.

As for specific help relating to k8s deployments, I'm afraid that given the complexities and customizations that each self hosting setup has, it is near impossible to provide a one-size-fits-all guidance beyond the compose file, which itself provides a solid foundation to build a k8s config off. 

The postgres backend was specifically built for local development in mind, it does not make sense to store your logs on your production database due to lack of separation of concerns, and it would be advisable to use the BQ backend instead. 
This is already advised in the [self hosting documentation](https://supabase.com/docs/reference/self-hosting-analytics/introduction).

As for postgres extension enabling, the supabase postgres image should have the extensions pre-installed, so i'm not quite sure what  you mean.

**Comment by Songyawn** [comment]
still same error .....

**Reply by Madjarx** [reply]
TLDR & ANSWER: Find the `analytics` service in the main `docker-compose.yml`. Shoud say something like `analytics` and `image: supabase/logflare:1.4.0` or whatever is the image version at the time of reading

Just bump up the version to 1.8.9 and you should see stuff being visible in your supabase dashboard

logs are being captured but for whatever reason you cant access them with the 1.4.0

Also generate an access token on the logflare website, its pretty straightforward, and use it for the `LOGFLARE_API_KEY` in the `.env`

**Reply by BeppeMarnell** [reply]
yep, me too!

**Comment by ilnytskyi** [comment]
Updating analytics image to 1.8.9 seem to fix the problem

**Comment by dangnhdev** [comment]
FYI: As of today the logflare version in docker-compose file is 1.12.5 but we still set version 1.8.9 to make this work.

**Reply by Ziinc** [reply]
You would not need to do so with the higher version.

**Reply by dangnhdev** [reply]
Sorry, I wasn’t very sober when I wrote the comment above. What I meant was: "**_I need to set that version to make it work_**" (referring to the latest Supabase Docker Compose). After an hour of trying everything to fix the 401 error in the Studio > Analytics dashboard, simply setting the Logflare image version to 1.8.9 made it work. I was too tired to try another version since there are still many issues to fix with this self-hosted stack 😅

**Reply by bcdon** [reply]
I'm also experiencing the same issue. The Logflare version in the docker-compose file is 1.12.5, and Studio > Analytics shows a 401 error. However, when I switch to version 1.8.9, the logs display properly.

**Reply by sosaurizar** [reply]
I can confirm this issue. I also had to downgrade from the default 1.12.5 in the docker-compose.yaml to 1.8.9 to get it working.

My setup is using the default docker-compose.yaml and the default .env file with the demo credentials.

The only configuration change I made was to enable the Docker socket for rootless Docker.
following the selfhost docker guide.

> If you are using rootless docker, edit .env and set DOCKER_SOCKET_LOCATION to your docker socket location. For example: /run/user/1000/docker.sock. Otherwise, you will see an error like container supabase-vector exited (0).

**Reply by cgimenes** [reply]
Same issue here. But I'm using Supabase CLI to run a local instance.

**Reply by JLAcostaEC** [reply]
I also experienced the same problem, and it was solved by changing the version to the one indicated by @dangnhdev

**Reply by AntonOfTheWoods** [reply]
I refuse to believe the maintainer is so incompetent to not know what is going on here. It is *obviously* because of Supabase's attitude towards development - it's just normal, non FOSS development done in public repos, with FOSS licences. They simply don't care about any uses of the various apps outside their SAAS context though, and the almost total lack of usage docs is proof of that. Half of the repo docs point to end user docs for their SAAS products FFS!

We can't really complain though... Because it *is* free (as in beer).

**Comment by cgimenes** [comment]
Workaround for those using Supabase CLI with `supabase start`:
```shell
docker pull supabase/logflare:1.8.9
docker rmi public.ecr.aws/supabase/logflare:1.12.5
docker tag supabase/logflare:1.8.9 public.ecr.aws/supabase/logflare:1.12.5
```

**Comment by onaralili** [comment]
This is still an issue with the latest stable version.

**Reply by JLAcostaEC** [reply]
Yes, I have also tried to update the images manually to the latest available ones (They are not the same as those described in the yaml) and it does not work.

**Comment by szjozsef** [comment]
It seems the latest working version (for me) is 1.12.0
even if I replace the deprecated LOGFLARE_API_KEY (with, or set additionally): LOGFLARE_PUBLIC_ACCESS_TOKEN
with any supabase/logflare 1.12.1+ version I receive the error code: 401 in self-hosted supabase dashboard (https://<supabase url>/project/default/logs/edge-logs

**Reply by dangnhdev** [reply]
Yes, you can see that the developers have updated the Logflare version in the docker-compose file to this version as well.

**Comment by soosap** [comment]
I would also like to chip into this discussion to confirm that none on the recent versions of the logflare image work with studio. Like everyone else I am receiving the 401 errors.

I can also confirm that v1.8.9 actually DOES work in my Kubernetes environment. It's is the ONLY version that does not return the 401 errors.

Kindly requesting @Ziinc to mark this discussion as not answered.

**Comment by cardilloscreations** [comment]
I'm running a self-hosted instance and I'm kinda amazed that this problem had gone unresolved for so long. I recently upgraded other components to the latest in the docker stack and now a rollback of `logflare` to `v1.8.9` is not even working.

[jeffmccune](https://github.com/jeffmccune) Do you have the ability to mark the question unanswered? There are a lot of folks on this thread so it has good visibility, but the only real answer that works is not really an answer.

**Comment by szjozsef** [comment]
It is somewhat working using the latest images (even supabase/logflare:1.15.3) just you have to :
- LOGFLARE_PUBLIC_ACCESS_TOKEN= (32 char random string)
- LOGFLARE_PRIVATE_ACCESS_TOKEN= (a different 32 char random string)
- in the docker-compose.yml use the LOGFLARE_PUBLIC_ACCESS_TOKEN for the services: analytics, vector
- in the docker-compose.yml use the LOGFLARE_PRIVATE_ACCESS_TOKEN for the services: studio, analytics
- use the new volumes/logs/vector.yml
- use  at least supabase/studio:2025.06.02-sha-8f2993d or newer

it failed for me with longer token (this is not well documented)

**Reply by cardilloscreations** [reply]
I'll verify later tonight, but this was all setup previously and worked when I rolled back to `v1.8.9`. Then I recently updated my images to the latest versions, but left `logflare` at `v1.8.9` and it stopped working again. Then I tried to update to `logflare` to `v1.15.3` and still no luck.

I do tend to have long tokens per typical security standards. Perhaps that is the issue. I'll report back if I have any success.

**Reply by cardilloscreations** [reply]
I was able to use my 64 character tokens without a problem once I picked up the new changes that were just recently published. See PR #36169. I'll post the details in a new comment.

**Comment by cardilloscreations** [comment]
I'm sharing this update for the benefit of anyone that comes across this post after struggling to get logging/analytics working with Supabase using the latest docker images. Apparently there were some changes made to LogFlare that didn't get updated in the self-hosted docker config until recently.

After studying PR #36169, and applying the following changes, everything is fully functional again, and I'm using the latest docker images as of June 17, 2025. This includes `supabase/logflare:1.15.3` and `supabase/studio:2025.06.16-sha-c4316c3`, etc.

**Here's what I did based on my review of PR #36169:**
1. Sync with the latest changes published. For me, this was just a `git fetch` and `git pull` to get the changes to the docker configuration. If you don't do that, you'll have to review carefully because there were changes to a few places.
2. Change your `.env` based on the changes in the `.env.example` from the PR. Mainly, removing the old env and adding the new PUBLIC and PRIVATE tokens. Others have said this is limited to 32 characters, but I used 64 character tokens, and it worked just fine.
3. Make sure you're using the latest updates. I like to do this by using the docker override file (`docker-compose.override.yml`) to just override the versions, so I don't have to touch the original version from the repo.
4. To pick up the new configuration, cycle down and up using `docker compose down` and `docker compose up -d`.
6. Everything worked as expected. :tada:

**Reply by roo12312** [reply]
still not working...

**Reply by roo12312** [reply]
with `supabase/studio:2025.06.16-sha-c4316c3`

its working

**Reply by georgiev-anton** [reply]
not working in k8s(

```analytics:
enabled: true
image:
repository: supabase/logflare
tag: "1.15.3"
environment
    LOGFLARE_SINGLE_TENANT: true
    LOGFLARE_SUPABASE_MODE: true
    LOGFLARE_PUBLIC_ACCESS_TOKEN: "...32 length token..."
    LOGFLARE_PRIVATE_ACCESS_TOKEN: "...32 length token..."
 ```

```
studio:
  enabled: true
  image:
    repository: supabase/studio
    tag: "2025.06.16-sha-c4316c3"
```



<img width="555" height="441" alt="image" src="https://github.com/user-attachments/assets/a9a89c11-f9ad-430e-b0d0-f7572f395c1b" />

**Reply by JLAcostaEC** [reply]
> not working in k8s(

Try with `latest` image version for studio and logflare

**Reply by georgiev-anton** [reply]
> > not working in k8s(
> 
> Try with `latest` image version for studio and logflare

same error(  Unauthorized

```
  image:
    repository: supabase/logflare
    pullPolicy: IfNotPresent
    tag: "dev-c8891e9"
```

```
  image:
    repository: supabase/studio
    pullPolicy: IfNotPresent
    tag: latest
 ```

**Reply by aantti** [reply]
@luizfelmach Jfyi, re https://github.com/supabase-community/supabase-kubernetes/
